central/delegatedregistryconfig/delegator/delegator.go

```diff
@@ -124,23 +123,18 @@ func (d *delegatorImpl) inferNamespace(ctx context.Context, imgName *storage.Ima
 	// Extract namespace from image path following OCP integrated registry convention.
 	namespace := utils.ExtractOpenShiftProject(imgName)
 
-	q := search.NewQueryBuilder().AddExactMatches(search.Namespace, namespace).AddExactMatches(search.ClusterID, clusterID).ProtoQuery()
+	defer centralMetrics.SetFunctionSegmentDuration(time.Now(), "ScanDelegatorInferNamespace")
 
-	defer centralMetrics.SetFunctionSegmentDuration(time.Now(), "ScanDelegatorSearchNamespaces")
-
-	// SearchNamespaces will only return a result if user has access.
-	namespaces, err := d.namespaceDS.SearchNamespaces(ctx, q)
+	namespaces, err := d.namespaceSACHelper.GetNamespacesForClusterAndPermissions(ctx, clusterID, nil)
```

Comment:
So, while you may drop the `Detection` one since it's a global resource and doesn't affect whether a cluster has implicit access or not (seems like `Detection` anyways isn't required for the image scan APIs), this should still pass `Image` as the requested permission. While it's true that the caller _does_ have to have `WRITE Image` to be able to call the API in the first place, this might leak namespaces.

The sac helper will retrieve all clusters where the current context has permissions for the specified resources. If left nil, a namespace will be returned if _any_ implicit resource access is given for a particular cluster.

Given an access scope of the following:
```yaml
cluster: remote
- namespace: something-else
- READ NetworkGraph
- namespace: default
  permissions:
  - READ Image
```

This would currently return the namespace `default` and `something-else` namespaces, even though the user isn't necessarily authorized to access `Image` within the `something-else` namespace. Hence, in this particular case, `Image` should be added to the requested permissions so ensure we only return namespaces which _have_ the requested permission.

Note that this also applies to the `GetClustersForPermissions` call, where the same principle applies (currently, we will return a list of _all_ clusters, but ideally would want to do this only for clusters where an implicit access to `Image` is given.

Reply:
https://github.com/stackrox/stackrox/pull/8016/commits/7ad931d2f71a26d7ec4603adfa8ab61533d6528c